doc/STYLE.md

```diff
@@ -0,0 +1,159 @@
+# Care And Feeding of Your Fellow Coders
```

Comment:
Maybe link also from `HACKING.md` to `STYLE.md`?

Reply:
Thanks, done.

Reply:
`HACKING.md` gets installed `$(docdir)`, usually `/usr/local/share/doc/c-lightning`.  If `HACKING.md` refers to `STYLE.md` perhaps it should be added to `DOC_DATA` in the main `Makefile`